public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -120,4 +136,38 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TraceToLo
   getQueryDisplayText(query: TempoQuery) {
     return query.query;
   }
+
+  private queryServiceMapPrometheus(request: DataQueryRequest<PromQuery>) {
+    return from(getDatasourceSrv().get(this.serviceMap!.datasourceUid)).pipe(
```

Comment:
Is serviceMap guaranteed to be defined?

Reply:
Basically yes cause it's checked before this method gets called, but yeah it's a bit implicit. I refactored it a bit to be more explicit about it.